src/org/traccar/BaseProtocolDecoder.java

```diff
@@ -31,6 +31,22 @@
 
     private final Protocol protocol;
 
+    public long addUnknownDevice(String uniqueId, SocketAddress remoteAddress) {
+        long deviceId = 0;
+        Device newDevice = new Device();
+        newDevice.setName(uniqueId);
+        newDevice.setUniqueId(uniqueId);
+
+        try {
+            Context.getDeviceManager().addDevice(newDevice);
+            Log.info("Nice meeting new device " + uniqueId + ".");
```

Comment:
Please use more formal language for log message. Something like "Automatically registered device XXX". Also, we don't use dots at the end of log messages, so it would nice to be consistent in that sense as well.